web/packages/teleterm/src/ui/services/connectionTracker/types.ts

```diff
@@ -50,10 +56,18 @@ export interface TrackedKubeConnection extends TrackedConnectionBase {
   kubeUri: KubeUri;
 }
 
+export interface TrackedAppConnection extends TrackedConnectionBase {
+  kind: 'connection.app';
+  appUri: AppUri;
```

Comment:
I feel like this should be called `targetUri`. It's called `kubeUri` for the kube connection only because of backwards compatibility when there was no underlying gateway.